galore/cross_sections.py

```diff
@@ -281,3 +281,261 @@ def _eval_fit(energy, coeffs):
                 orb: _eval_fit(energy, np_fromstr(coeffs))
                 for orb, coeffs in orbitals_fits}})
     return el_cross_sections
+
+
+
+import tarfile
+import numpy as np
+
+
+def read_csv_file(tar_file_name, file_path):
+    """
+    Generate data from csv archive without decompression
+
+    Args:
+        tar_file_name (str): path to tarfile of CSV data
+        file_path(str): path to individual CSV file within tarfile
+
+    Returns:
+        dict: containing 'headers', 'electron_counts' 
+        (lists of str and int respectively) and 'data_table', 
+        a 2-D nested list of floats. Missing data is represented as None.
+
+    """
+
+    # Open zipfile
+    with tarfile.open(tar_file_name) as tf:
+        with tf.extractfile(file_path) as hello:
+            # get data as string
+            data = hello.read().decode()
+    # string to list
+    data_string = data.split('\r\n')
+
+    # get number of colunm headers
+    colunm_headers = [colunm_header for colunm_header in data_string[0].split(
+        ',') if colunm_header != '']
+    lenth = len(colunm_headers)
+
+    # build main matrix
+    main_matrix = []
+    rows = range(len(data_string))
+    for row in rows:
+        data_each_row = data_string[row].split(',')[0:lenth]
+        main_matrix.append(data_each_row)
+
+    # build cross sections table
+    empty_value = main_matrix[-2]
+    # remove empty values
+    midterm = [row for row in main_matrix if row != empty_value]
+    data_table = midterm[1:-2]
+    electron_counts_list = midterm[-2]
+    electron_counts = [i for i in electron_counts_list if i != ''][1:]
+    # build result dict
+    result_dict = {}
+    result_dict['headers'] = colunm_headers
+    result_dict['electron_counts'] = electron_counts
+    result_dict['data_table'] = data_table
+
+    return result_dict
+
+
+def get_avg_orbital_cross_sections(subshells_cross_sections, electrons_number):
```

Comment:
This is a good candidate to be a "private" function with a leading underscore (e.g. `def _get_avg_orbital_cross_sections`), because:
- it is a bit sensitive (has "funny" input requirements)
- it is likely to changed, moved or removed as this code is further developed
- we don't expect other users or libraries to call it; it is meant to be called inside this module

Generally when functions are "public" (which in Python just means they don't have an underscore at the start of the name) we are obliged to look after them and avoid breaking the compatibility without warning. So we limit the number that are "public" and try to make those ones more robust and general-purpose.